src/Bicep.Core/Syntax/ImportSpecification.cs

```diff
@@ -46,35 +49,57 @@ private ImportSpecification(string name, string version, bool isValid, TextSpan
 
         public static ImportSpecification From(SyntaxBase specificationSyntax)
         {
-            switch (specificationSyntax)
+            return specificationSyntax switch
             {
-                case StringSyntax stringSyntax when stringSyntax.TryGetLiteralValue() is { } value:
-                    var (name, version, isValid) = Parse(value);
-                    var span = isValid ? new TextSpan(stringSyntax.Span.Position + 1, name.Length) : stringSyntax.Span;
-
-                    return new ImportSpecification(name, version, isValid, span);
-
-                case SkippedTriviaSyntax trivia:
-                    return new ImportSpecification(trivia.TriviaName, trivia.TriviaName, false, trivia.Span);
+                StringSyntax stringSyntax when stringSyntax.TryGetLiteralValue() is { } value
+                    => CreateFromStringSyntax(stringSyntax, value),
+                SkippedTriviaSyntax trivia
+                    => new ImportSpecification(
+                        trivia.TriviaName,
+                        trivia.TriviaName,
+                        trivia.TriviaName,
+                        false,
+                        trivia.Span),
+                _
+                    => new ImportSpecification(
+                        LanguageConstants.ErrorName,
+                        LanguageConstants.ErrorName,
+                        LanguageConstants.ErrorName,
+                        false,
+                        specificationSyntax.Span)
+            };
+        }
 
-                default:
-                    return new ImportSpecification(LanguageConstants.ErrorName, LanguageConstants.ErrorName, false, specificationSyntax.Span);
+        public StringSyntax ToPath()
+        {
+            if (!this.IsValid)
+            {
+                throw new InvalidOperationException("Cannot convert invalid import specification to path.");
```

Comment:
I don't think you want to throw here; it'll be common for invalid or incomplete templates to have an invalid specification. Could you return a nullable value (or maybe a [`Result`](https://github.com/Azure/bicep/blob/main/src/Bicep.Core/Utils/Result.cs))?

Reply:
Great catch! In addition to `Result` - there's also [ResultWithDiagnostic](https://github.com/Azure/bicep/blob/9899a47c48e76b88d326846e447b3c2162e8f63a/src/Bicep.Core/Diagnostics/ResultWithDiagnostic.cs#L8) which allows you to return diagnostics about the problem.

Reply:
Yes, addressed accordingly